File: backend/internal/model/vo/monitor.go
package vo

// CacheStatsVO represents basic Redis/cache statistics for monitoring.
type CacheStatsVO struct {
	RedisVersion     string  `json:"redis_version"`
	Mode             string  `json:"mode"`
	UsedMemory       int64   `json:"used_memory"`
	UsedMemoryHuman  string  `json:"used_memory_human"`
	DBSize           int64   `json:"db_size"`
	ConnectedClients int64   `json:"connected_clients"`
	KeyspaceHits     int64   `json:"keyspace_hits"`
	KeyspaceMisses   int64   `json:"keyspace_misses"`
	HitRate          float64 `json:"hit_rate"`
}

// OnlineUserVO represents an online user session for monitoring.
type OnlineUserVO struct {
	SessionID      string `json:"session_id"`
	UserCode       string `json:"user_code"`
	Username       string `json:"username"`
	DeptName       string `json:"dept_name"`
	IP             string `json:"ip"`
	LoginLocation  string `json:"login_location"`
	HostIP         string `json:"host_ip"`
	UserAgent      string `json:"user_agent"`
	OS             string `json:"os"`
	Browser        string `json:"browser"`
	BrowserVersion string `json:"browser_version"`
	BrowserDetail  string `json:"browser_detail"`
	LoginTime      int64  `json:"login_time"`
	LastActive     int64  `json:"last_active"`
}

// DatabaseStatsVO represents database health and basic schema statistics.
type DatabaseStatsVO struct {
	Status          string `json:"status"`
	Driver          string `json:"driver"`
	Database        string `json:"database"`
	Version         string `json:"version"`
	TableCount      int64  `json:"table_count"`
	OpenConnections int    `json:"open_connections"`
	InUse           int    `json:"in_use"`
	Idle            int    `json:"idle"`
}

File: frontend/admin-web/src/services/antd-gin-api/monitor.ts
import { request } from "@umijs/max";

export interface CacheStats {
  redis_version?: string;
  mode?: string;
  used_memory?: number;
  used_memory_human?: string;
  db_size?: number;
  connected_clients?: number;
  keyspace_hits?: number;
  keyspace_misses?: number;
  hit_rate?: number;
}

export interface OnlineUser {
  session_id: string;
  user_code: string;
  username: string;
  dept_name?: string;
  ip?: string;
  login_location?: string;
  host_ip?: string;
  user_agent?: string;
  os?: string;
  browser?: string;
  browser_version?: string;
  browser_detail?: string;
  login_time?: number;
  last_active: number;
}

export interface DatabaseStats {
  status?: string;
  driver?: string;
  database?: string;
  version?: string;
  table_count?: number;
  open_connections?: number;
  in_use?: number;
  idle?: number;
}

/** 获取缓存监控数据 */
export async function getCacheStats(opts?: { skipErrorHandler?: boolean }) {
  return request<{
    code: number;
    data: CacheStats;
    message?: string;
  }>("/api/v1/system/monitor/cache", {
    method: "GET",
    ...opts,
  });
}

/** 获取数据库监控数据 */
export async function getDatabaseStats(opts?: { skipErrorHandler?: boolean }) {
  return request<{
    code: number;
    data: DatabaseStats;
    message?: string;
  }>("/api/v1/system/monitor/database", {
    method: "GET",
    ...opts,
  });
}

/** 获取在线用户列表 */
export async function getOnlineUsers(opts?: { skipErrorHandler?: boolean }) {
  return request<{
    code: number;
    data: OnlineUser[];
    message?: string;
  }>("/api/v1/system/monitor/online-users", {
    method: "GET",
    ...opts,
  });
}

/** 强退在线用户 */
export async function forceLogoutOnlineUser(sessionId: string) {
  return request<{
    code: number;
    data: { forced: boolean };
    message?: string;
  }>(`/api/v1/system/monitor/online-users/${sessionId}`, {
    method: "DELETE",
  });
}
